service/history/workflow/mutable_state_impl.go

```diff
@@ -451,6 +451,12 @@ func (e *MutableStateImpl) UpdateCurrentVersion(
 
 	if state, _ := e.GetWorkflowStateStatus(); state == enumsspb.WORKFLOW_EXECUTION_STATE_COMPLETED {
 		// do not update current version only when workflow is completed
```

Comment:
It sounds like the new comment contradicts this one? Should this line be removed?